manta-accounting/src/transfer/canonical.rs

```diff
@@ -738,4 +738,43 @@ where
     {
         self.reconstruct_utxo(parameters, address).eq(utxos)
     }
+
+    /// Returns the [`TransferShape`] of `self`.
+    #[inline]
+    pub fn shape(&self) -> TransferShape {
+        match self {
+            Self::ToPrivate(_, _) => TransferShape::ToPrivate,
+            Self::PrivateTransfer(_) => TransferShape::PrivateTransfer,
+            Self::ToPublic(_, _) => TransferShape::ToPublic,
+        }
+    }
+
+    /// Verifies `self` against `transferpost`.
+    #[inline]
+    pub fn verify(
+        &self,
+        parameters: &Parameters<C>,
+        address: &Address<C>,
+        transferpost: TransferPost<C>,
+    ) -> bool
+    where
+        Utxo<C>: PartialEq,
+    {
+        if !TransferShape::from_post(&transferpost)
```

Comment:
Does this actually verify the proofs? Doesn't it need the checks found here: https://github.com/Manta-Network/manta-rs/blob/6824ecbfaa4fbb7a7fa9b5fb558d8985954afb17/manta-accounting/src/transfer/mod.rs#L2199

Otherwise I could just make a phony `TransferPost<C>` and as long as UTXO is onchain it would seem valid

Reply:
But the `TransferPost` is the challenge/public input provided by the verifier. It can be as phony as you want, then the prover will simply not be able to generate a valid proof. What needs to be verified is the `TransactionData`, and in that case, the verifier challenges the prover with a `TransferPost` and receives `TransactionData` from the verifier. You then verify it against the `TransferPost` you sent at the beginning. 

`identity_verification` works differently. The verifier provides the `IdentifiedAsset` and the prover generates a `TransferPost`. In this case, what needs to be verified is the `TransferPost` itself.